uefi-logger/src/writer.rs

```diff
@@ -37,27 +38,33 @@ impl fmt::Write for OutputWriter {
             let mut add_char = |ch| {
                 // UEFI only supports UCS-2 characters, not UTF-16,
                 // so there are no multibyte characters.
-                let ch = ch as u16;
-
                 buf[i] = ch;
-
                 i += 1;
 
                 if i == BUF_SIZE {
-                    flush_buffer(&mut buf, &mut i)
+                    match flush_buffer(&mut buf, &mut i) {
```

Comment:
I think here we can use a `Result` [combinator](https://doc.rust-lang.org/std/result/enum.Result.html):
```rust
flush_buffer(&mut buf, &mut i).map_err(|_| Status::ProtocolError)
```